### DISCUSSION TITLE
Is it possible to assign a Dataview query to a menu item?

### DISCUSSION THREAD

**Discussion by kenNash** [chrisgurney/obsidian-note-toolbar#70] [Ideas]
Hi guys.

Is it possible to define a menu item in terms of a Dataview inline query?

I'd like to turn this navigation section

```
`= "[[" + string(date(this.file.day) - dur(1 year)) + "]]"`
`= "[[" + string(date(this.file.day) - dur(1 month)) + "]]"`
`= "[[" + string(date(this.file.day) - dur(1 week)) + "]]"`
`= "[[" + string(date(this.file.day) - dur(1 day)) + "]]"` ← **`= this.file.name`** → `= "[[" + string(date(this.file.day) + dur(1 day)) + "]]"`
`= "[[" + string(date(this.file.day) + dur(1 week)) + "]]"`
`= "[[" + string(date(this.file.day) + dur(1 month)) + "]]"`
`= "[[" + string(date(this.file.day) + dur(1 year)) + "]]"`
```

into a Note Toolbar. Is it possible?

**Comment by chrisgurney** [comment]
@kenNash While you can't use Dataview in a toolbar, you can use it in a `note-toolbar` callout.

Here's an example of a Dataview query in a callout that shows your bookmarks, for example:

```
> [!note-toolbar|even]
> ```dataview
> LIST 
> WHERE file.starred
> SORT file.name
> ```
```

So I believe you should be able to create a bulleted list with the items you identified, or output a code block from a dataviewjs block.

**Reply by chrisgurney** [reply]
There's been more than one request for Dataview support in this plugin, but I want to wait until the Obsidian team has progressed with the possibly similar functionality they have in the roadmap. 

Follow [this issue](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/64) for updates on the plugin specifically.

**Comment by kenNash** [comment]
Thank you, @chrisgurney, for your prompt and kind answer! 🙏🏽

I'll wait.

**Comment by chrisgurney** [comment]
@kenNash The latest [beta release](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/171) adds support for Dataview execution from buttons.

While this doesn't solve your particular use case, I will _also_ be looking into the possibility of allowing Dataview expressions for labels, links, and such. So hopefully keep an eye out for that!

**Reply by kenNash** [reply]
Oh, thank you!

**Comment by chrisgurney** [comment]
@kenNash I'm closing this discussion. But please let me know if you've got this working, or if not, if you read the [user guide page](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Dataview) and still have questions.
